hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegion.java

```diff
@@ -2047,6 +2049,16 @@ public Configuration getReadOnlyConfiguration() {
     return new ReadOnlyConfiguration(this.conf);
   }
 
+  @Override
+  public int getMinBlockSizeBytes() {
+    if (minBlockSizeBytes > 0) {
+      return minBlockSizeBytes;
+    }
+    minBlockSizeBytes = Arrays.stream(this.htableDescriptor.getColumnFamilies())
+      .mapToInt(ColumnFamilyDescriptor::getBlocksize).min().orElse(HConstants.DEFAULT_BLOCKSIZE);
+    return minBlockSizeBytes;
+  }
```

Comment:
This is lazily evaluated and cached because it will be used in the hot path of requests as they evaluate their quotas

Reply:
I might recommend simply calculating this in HRegion construction. It's basically free to do there, and in terms of memory usage a 0 value takes the same as any other.

Otherwise the variable needs to be volatile and you need to handle synchronization

Reply:
Good thinking